dev/RadioButtons/ColumnMajorUniformToLargestGridLayout.cpp

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "ResourceAccessor.h"
+#include "Utils.h"
+#include "ColumnMajorUniformToLargestGridLayout.h"
+
+CppWinRTActivatableClassWithBasicFactory(ColumnMajorUniformToLargestGridLayout);
+
+// IItemsControlOverrides
+
+ColumnMajorUniformToLargestGridLayout::ColumnMajorUniformToLargestGridLayout()
+{
+}
+
+winrt::Size ColumnMajorUniformToLargestGridLayout::MeasureOverride(
+    winrt::NonVirtualizingLayoutContext const& context,
+    winrt::Size const& availableSize)
+{
+    if (auto const children = context.Children())
+    {
+        auto const maxColumns = std::max(1, MaximumColumns());
+        MUX_ASSERT(maxColumns > 0);
+        auto const maxItemsPerColumn = static_cast<int>(std::ceil(static_cast<double>(children.Size()) / static_cast<double>(maxColumns)));
+
+        for (auto const child : children)
+        {
+            child.Measure(availableSize);
+        }
+        auto const largestChildSize = LargestChildSize(context);
```

Comment:
you could calculate this in the loop above and avoid iterating through all the chidren multiple times. You could also save it in your layout (or layoutstate) and use that in Arrange and avoid an extra walk there as well.